### PR TITLE
Remove CRSP V1 residuals and express common-stock and NYSE screens na…

### LINKED ISSUE #202: Remove reconstructed CRSP V1 `shrcd` from the pipeline (finish the CIZ/V2 migration)
### Problem

The pipeline is fully on CRSP V2 (CIZ) for data sourcing, but still carries a CRSP V1 leftover: `gen_crsp_sf` reconstructs the legacy numeric `shrcd` from CIZ descriptor fields purely so downstream V1-era code keeps working. This is a compatibility shim, not a real variable — CIZ has no `shrcd`. We should finish the migration and express the universe screen natively in CIZ, dropping `shrcd` (and the other reconstructed V1 vars) entirely.

Keeping a synthesized V1 column is misleading and lossy:
- The CIZ→shrcd map only ever emits `10` (or null), so the downstream `shrcd IN (10,11,12)` screen is silently just `== 10`. The 11/12 branches are dead, and the old 10/11/12 semantics (US- vs foreign-incorporated, special status) are collapsed — that information lives in CIZ fields (`issuertype`, incorporation) but is thrown away.
- `crsp_shrcd` / `crsp_exchcd` are carried as V1 diagnostics that no longer mean what their names imply.

### Proposed change

Remove `shrcd` from the pipeline and define the common-stock screen directly from CIZ fields.

- **`gen_crsp_sf`** (`aux_functions.py:529`) — stop emitting `shrcd` (and reconstructed `exchcd`); keep the CIZ source fields (`securitytype`, `securitysubtype`, `sharetype`, `issuertype`, `primaryexch`, `conditionaltype`) instead.
- **`combine_crsp_comp_sf`** (`aux_functions.py:2813`) — derive `common` directly from the CIZ predicate (`EQTY & COM & NS & issuertype ∈ {ACOR, CORP}`) rather than `shrcd IN (10,11,12)`; drop the `crsp_shrcd` / `crsp_exchcd` V1 passthrough columns.
- Audit the `common`-flag consumers so nothing breaks: `_main_filter_expr` / `filter_dsf` / `filter_msf` / `filter_world`, `nyse_size_cutoffs`, `return_cutoffs`, `apply_stock_filter_and_compute_indexes`, `load_mkt_equity_data`, `ap_factors`, `mispricing_factors`, `quality_minus_junk`.
- Grep for any remaining `shrcd` / `crsp_shrcd` / `crsp_exchcd` / `exchcd` references and remove or replace with CIZ equivalents.

### Acceptance criteria

- No `shrcd` (or other reconstructed CRSP V1 variable) anywhere in `src/jkp/data/`.
- `common` / `exch_main` defined natively from CIZ fields, behavior-preserving.
- Equivalence test: pipeline output (universe membership, `common`, size groups) identical before vs after on a sample.
- Tests + lint pass.

### Notes

Assigning @miahdy — she did the original CIZ/V2 migration and knows the field mapping.

### PR DESCRIPTION
**Pull request by miahdy**
## Summary

Closes #202.

Finish the CRSP V2 (CIZ) migration by removing synthesized V1 compatibility columns (`shrcd`, `exchcd`, `crsp_shrcd`, `crsp_exchcd`) and defining universe screens directly from native CIZ fields.

- **`gen_crsp_sf`**: emit `common`, `primaryexch`, and `conditionaltype` instead of reconstructed `shrcd`/`exchcd`
- **`prepare_crsp_sf`**: NASDAQ volume adjustment uses `(primaryexch == "Q") & (conditionaltype == "RW")`
- **`combine_crsp_comp_sf`**: read `common` from upstream; pass through `primaryexch`/`conditionaltype` in monthly world output
- **NYSE logic** (`nyse_size_cutoffs`, `sort_ff_style`, `ap_factors`, `portfolios`): replace `crsp_exchcd == 1` with `primaryexch == "N"` and `conditionaltype == "RW"` on CRSP rows; Compustat NYSE remains `comp_exchg == 11`
- **Tests & docs**: update unit/portfolio fixtures and `documentation/documentation.tex` / release notes

## Note

The old `shrcd IN (10, 11, 12)` screen was effectively `shrcd == 10` only (the shim never emitted 11 or 12). The new `common` flag and NYSE/NASDAQ predicates are logically equivalent to the prior reconstructed codes, so universe membership should be unchanged.

## Equivalence check

Compared `data/processed/portfolios/hml.parquet` (new) against `data-old/processed/portfolios/hml.parquet` (old), plus company-level characteristics spot checks.

### `hml.parquet` (portfolio returns)

Identical keys: 2,712,115 rows, 71 countries, 153 characteristics, date range 1926-01 to 2025-12. Zero key mismatches.

| Series | Correlation | Exact match | Max \|diff\| |
|--------|-------------|-------------|--------------|
| `ret_ew` | 0.9999996 | 99.10% | 0.046 |
| `ret_vw` | 0.9999994 | 98.80% | 0.054 |
| `ret_vw_cap` | 0.9999994 | 98.88% | 0.054 |

Every country's minimum correlation across all three return series is > 0.9999. Worst countries are small markets (NZL, ZAF, GRC); USA is effectively bit-identical (max \|diff\| ≈ 2e-5).

~1.2% of rows have any non-zero diff, but almost all are floating-point noise: only 15 rows have \|diff\| > 1e-10, and only 6 rows have \|diff\| > 1e-2. All 10 rows with \|ret_vw diff\| > 1e-4 are in NZL, ZAF, or GRC across four characteristics (`iskew_ff3_21d`, `ni_ivol`, `dolvol_126d`, `dolvol_var_126d`) — thin-stock edge cases, not systematic drift.

Non-return columns: `n_stocks` mismatches = 6 (all ±1 stock); `signal` diffs = 13,511 (almost entirely tiny `dolvol_126d` float noise; overall corr = 1.0).

### Characteristics (company-level spot check)

Expected schema change from CIZ migration: removed `crsp_shrcd`/`crsp_exchcd`; added `conditionaltype`/`primaryexch`. Row membership unchanged (e.g. USA `common` flag: 0 changes).

| Country | Stock-level diffs (meaningful) |
|---------|-------------------------------|
| USA | Only 6 chars with any \|diff\| > 1e-6 (`dolvol_126d`, `zero_trades_*`); max 2.7e-5 |
| NZL | All focus chars identical at stock level |
| ZAF | `iskew_ff3_21d` only real diff (corr 0.9999, 128 rows > 1e-6) |
| GRC | All focus chars identical at stock level |

10-company USA subset: all characteristics match within machine precision.

**Bottom line:** new pipeline output matches old extremely closely. Return correlations > 0.999999 overall; meaningful differences isolated to small-market edge cases with thin stock counts.

## Test plan

- [x] `uv run ruff check src/jkp/data/ tests/`
- [x] `uv run ruff format --check src/jkp/data/ tests/`
- [x] `uv run pytest tests/unit/` (659 passed)
- [x] Grep `src/` and `tests/` for `shrcd`, `exchcd`, `crsp_shrcd`, `crsp_exchcd` — zero hits
- [x] Equivalence check: `hml.parquet` return correlations > 0.999999; characteristics spot check (USA, NZL, ZAF, GRC) — see above
```

### CODE REVIEW

**Comment by FernandoRDLL** [comment]
Hello @miahdy 
How much time passed between both runs? One after the other or more?

File: src/jkp/data/aux_functions.py [hunk 1]
```diff
@@ -2214,7 +2210,7 @@ def combine_crsp_comp_sf(paths: DataPaths) -> None:
             COPY (
                 SELECT
                     id, permno, permco, gvkey, iid, excntry, exch_main, common,
-                    primary_sec, bidask, crsp_shrcd, crsp_exchcd, comp_tpci, comp_exchg,
+                    primary_sec, bidask, primaryexch, conditionaltype, comp_tpci, comp_exchg,
```
Comment (FernandoRDLL) [inline src/jkp/data/aux_functions.py]: This changes the schema of `data/processed/characteristics/` — drops `crsp_shrcd`/`crsp_exchcd`, adds `primaryexch`/`conditionaltype`. Per CLAUDE.md that needs a `CHANGELOG_DATA.md` entry. Could you add one under a new dated header referencing #205?

File: src/jkp/data/aux_functions.py [hunk 2]
```diff
@@ -2700,7 +2696,8 @@ def nyse_size_cutoffs(paths: DataPaths, data_path):
                 QUANTILE_DISC(me, 0.50)     AS nyse_p50,
                 QUANTILE_DISC(me, 0.80)     AS nyse_p80
             FROM self
-            WHERE  crsp_exchcd = 1
+            WHERE  primaryexch = 'N'
```
Comment (FernandoRDLL) [inline src/jkp/data/aux_functions.py]: `primaryexch = 'N' & conditionaltype = 'RW'` is now copy-pasted in four spots (here, `sort_ff_style`, `portfolios`, and the `ap_factors` lag list). The old `exchcd` kept this in one place; scattering it invites drift if someone edits one copy. Can we compute a single native-derived `crsp_nyse` flag once in `gen_crsp_sf` and reference it downstream? Still honors the "no reconstructed V1 codes" goal since it comes straight from the CIZ fields.

**Review by FernandoRDLL — CHANGES_REQUESTED** [review]
Nice cleanup — traced each rewritten screen back to the old one and they line up:

- `crsp_exchcd == 1` ↔ `primaryexch == 'N' & conditionaltype == 'RW'`, NASDAQ `exchcd == 3` ↔ `primaryexch == 'Q' & conditionaltype == 'RW'` — the old `exchcd` cases encoded exactly those.
- `common` is the same boolean the old `shrcd` shim used (`shrcd` was only ever 10 or null), so `common == 1` is unchanged. Non-common rows go `0` → `null`, but they're filtered out either way.
- bp_stock swap from `crsp_exchcd.is_null()` to `primaryexch.is_null()` on the Compustat branch is safe — `comp_exchg == 11` only fires on Compustat rows, where both are null.
- Columns flow through end to end, daily `ap_factors` reads its lag vars from monthly `world_data_prelim`, and grep shows zero `shrcd`/`exchcd`/`crsp_*` left in any `.py`. The >0.999999 equivalence matches a no-op.

Two things to fix before merge — left them inline where they land.

Rest LGTM once these two are in.

File: src/jkp/data/aux_functions.py (hunk 2 above)
Comment (FernandoRDLL) [reply src/jkp/data/aux_functions.py]: Double check this filter please

File: src/jkp/data/aux_functions.py (hunk 1 above)
Comment (FernandoRDLL) [reply src/jkp/data/aux_functions.py]: Double check this please

**Comment by FernandoRDLL** [comment]
Hey @miahdy 
I saw you made changes, in order for me to review them and be able to approve the PR you need to resolve the conversation. Normally, you put the changes in the code you made to address each thing.
Thank you for all your work!

**Comment by miahdy** [comment]
> Hello @miahdy How much time passed between both runs? One after the other or more?

Hi @FernandoRDLL , could you please clarify what exactly do you mean by "both runs"?

**Comment by miahdy** [comment]
> Hey @miahdy I saw you made changes, in order for me to review them and be able to approve the PR you need to resolve the conversation. Normally, you put the changes in the code you made to address each thing. Thank you for all your work!

Hi, sorry I wanted to double check the code before I resolve the conversations. It is now ready. Thank you for reviewing them!

**Comment by FernandoRDLL** [comment]
> > Hello @miahdy How much time passed between both runs? One after the other or more?
> 
> Hi @FernandoRDLL , could you please clarify what exactly do you mean by "both runs"?

I mean the base run and the new run that you used to create the diff table

**Comment by FernandoRDLL** [comment]
@miahdy Great job here too!

**Review by FernandoRDLL — APPROVED** [review]
Re-reviewed: CHANGELOG added; NYSE predicate deduplicated to a single native-derived crsp_nyse flag propagated end-to-end (production single-source; legacy-fixture copies intentionally independent). Equivalence >0.999999. CI green. LGTM.
